src/C-DEngine/C-DEngines/C-DThingService/cdeTheThing.cs

```diff
@@ -2093,13 +2138,15 @@ public void ClearAllEvents()
         public void RegisterEvent(string pEventName, Action<ICDEThing, object> pCallback)
         {
             if (pCallback == null || string.IsNullOrEmpty(pEventName)) return;
-            if (MyRegisteredEvents.ContainsKey(pEventName))
+            MyRegisteredEvents.AddOrUpdate(pEventName, (key) => new Action<ICDEThing, object>(pCallback), (key, existingAction) =>
```

Comment:
TheThing is derived from TheDataBase which also contains Event handling. Do you want to make the same changes there as well? Best would even be to somehow use the event management from TheDataBase and just route this to the base to avoid double code

Reply:
Just committed a new RegisteredEventHelper class and replaced all event handler logic.

Reply:
The events in TheThing and TheDataBase have different method signatures, so we can't completely unify them without breaking compatibility.